### ISSUE TITLE
Add option for setting now context

### ISSUE THREAD

**Issue by pauldix** [influxdata/platform#37]
The query engine executes with a specific time for `time.Now()`. This impacts any place where a query specifies relative times like `range(start:-1h)`. This option gives the user the ability to specify what that time should be.

```
option now {
  // time is the RFC3339 time a query should execute as
  time: 2006-01-02T15:04:05Z07:00
  // round a method for rounding to the nearest time. round and time are mutually exclusive
  round: 1h
}
```

**Comment by mark-rushakoff** [comment]
This was closed by mistake.

**Comment by lyondhill** [comment]
good catch. I forgot about the 'resolved' keyword

**Comment by mark-rushakoff** [comment]
I believe this was addressed by #388.
